## Cuarta actividad evaluativa

Profesor Alfonso Cervantes Barragán - 05-10-2020

Universidad Sergio Arboleda - Informatica III

### Descripción del contexto.

En esta actividad el estudiante debera realizar un análisis exploratorio de Datos (EDA) del dataset *religion-india.csv*, que contiene información sobre los estados en india y su población de prisioneros de los años 2001 al 2013.  

Información del Datasets:

|Columna |	Descripción |  
|--------|--------------|
|state_name 	|Nombre del estado|   
|is_state| Es un estado oficial de la India|   
|year |	año  | 
|gender|	genero|   
|convicts |	Orientación religiosa de los prisioneros|  
|under_trial| bajo juicio |  
|detenues |	detenciones |  
|others| otras razones|  


#### Entregable

El estudiante debera entregar un documento con extensión *ipynb* mediante el taller "Cuarta Evaluación " que se encuentra en el curso virtual Informatica III en el plataforma Sergio Virtual


#### Plazo

El plazo de entrega sera hasta el sabado, 24 de octubre a las 11:30 p.m. En el aula del curso en la plataforma Sergio Virtual estará disponible un taller en el cual deberan subir el archivo .ipynb (Jupyter notebook) con todo el código requerido. 

No esta permitido resolver los ejercicios a mano ni ingresar la respuesta sin que haya un código que lo respalde.

#### Donde puedo trabajar?

Este ejercicio se puede realizar en el entorno virtual creado con base en el repositorio github.com/alfonsocerv/datascience en la carpeta /ejericicio/caso10.

#### Criterios de evaluación

1. Todo el código requerido debe estar presente. Cada ejercicio debe tener su respectiva respuesta.
2. No pueden haber errores que se generen al ejecutar el código.
3. El código debe cumplir con las especificaciones de cada ejercicio.

### Ejercicio 1

Importa las librerias pandas, numpy y matplotlib

#### Respuesta

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
df = pd.read_csv("religion-india.csv")

#### Queremos saber de manera general de qué se trata esta base de datos

In [3]:
df.head()

state_name  is_state  year  gender   religion  convicts  under_trial  \
0  A & N Islands         0  2001  Female  Christian         0            0   
1  A & N Islands         0  2001  Female      Hindu         3            1   
2  A & N Islands         0  2001  Female     Muslim         0            0   
3  A & N Islands         0  2001  Female     Others         0            0   
4  A & N Islands         0  2001  Female       Sikh         0            0   

   detenues  others  
0         0       0  
1         0       0  
2         0       0  
3         0       0  
4         0       0

Podemos decir que es una base de datos que nos muestra por medio del año en que se realizó un censo, el tipo de religión y el Estado donde se encuentran ubicado, tambien nos muestra una columna muy importante, la cual determina si el estado es o no oficial en India, encontramos la información de los prisioneros de la India determinada por el género.
Registra si se encuentran detenidos o bajo juicio.

#### Queremos saber durante cuantos años se realizaron los sensos y en total, cuántos se realizaron

In [4]:
df["year"].nunique()

13

In [5]:
df.shape[0]

4460

En la India durante 13 años, se realizaron 4460 censos para obtener información sobre sus prisioneros.

#### Queremos saber cuántos tipos de religión hay en la base de datos

In [6]:
df["religion"].unique()

array(['Christian', 'Hindu', 'Muslim', 'Others', 'Sikh'], dtype=object)

Aquí nos muestran 5 tipos de religiones. Podemos observar que aunque el hinduismo es la religión ampliamente practicada en el Sur de Asia, no es la unica religión practicada por los prisioneros en la India. Practican otro tipo de religión o también se puede ver influenciado por prisioneros extranjeros


#### Queremos saber el número de mujeres y hombres que han estado bajo juicio durante los 13 años que se realizó el censo

In [7]:
df[["year","gender", "under_trial"]].groupby(["year", "gender"]).nunique() 

under_trial
year gender             
2001 Female           56
     Male            117
2002 Female           61
     Male            117
2003 Female           58
     Male            122
2004 Female           58
     Male            126
2005 Female           57
     Male            124
2006 Female           58
     Male            126
2007 Female           67
     Male            121
2008 Female           60
     Male            126
2009 Female           63
     Male            124
2010 Female           58
     Male            123
2011 Female           67
     Male            126
2012 Female           69
     Male            130
2013 Female           64
     Male            125

Podemos observar que en los trece años, el género masculino predominó en cantidad en comparación a las mujeres bajo jucio. Esto puede deberse a que la población masculina es mayor en toa la India y por eso se ve influenciado en los datos o porque las mujeres cumplen más las normas

#### Queremos saber el promedio de personas que a lo largo de los 13 años han estado bajo juicio

In [8]:
df["under_trial"].sum().mean()

3126505.0

Con esta información podemos determinar también un promedio por año, lo cual significaría un promedio de 2405000,38 personas que anualmente estuvieron bajo juicio en la India 

#### Queremos saber de las personas convictas en el estado de A & N Islands cuál es la religión que predomina

In [9]:
df[["state_name","convicts", "religion"]].groupby(["state_name","religion"]).sum().head(10)

convicts
state_name     religion           
A & N Islands  Christian       656
               Hindu           637
               Muslim          217
               Others         1593
               Sikh             10
Andhra Pradesh Christian      5302
               Hindu         49815
               Muslim         7475
               Others          764
               Sikh            176

Según una insvestigación previa es totalmente verdadero que en el estado de las Islas Andamán y Nicobar el tipo de relion que predomine sea *others* ya que en la isla no profesan una religión en sí, sino que ellos tienen creencias animistas. Creen  en espíritus de los bosques, el mar, la enfermedad y los ancestros. 

#### Queremos saber todos los estados estados que se encuentran en la India

In [10]:
df["state_name"].unique()

array(['A & N Islands', 'Andhra Pradesh', 'Arunachal Pradesh', 'Assam',
       'Bihar', 'Chandigarh', 'Chhatisgarh', 'Chhattisgarh',
       'D & N Haveli', 'Daman & Diu', 'Delhi', 'Goa', 'Gujarat',
       'Haryana', 'Himachal Pradesh', 'Jammu & Kashmir', 'Jharkhand',
       'Karnataka', 'Kerala', 'Lakshadweep', 'Madhya Pradesh',
       'Maharashtra', 'Manipur', 'Meghalaya', 'Mizoram', 'Nagaland',
       'Orissa', 'Pondicherry', 'Punjab', 'Rajasthan', 'Sikkim',
       'Tamilnadu', 'Tripura', 'Uttar Pradesh', 'Uttaranchal',
       'West Bengal'], dtype=object)

In [11]:
df["state_name"].nunique()

36

Existen 36 estados en la base de datos y notamos uno muy conocido, el cual es Delhi, ahi encontramos a nueva Delhi la cual es la capital de la India. Dentro de estos 36 estados, 28 son subdivisiones de estados de la India. Siete correspondenden a territorios de la union, los cuales se diferencian de los otros por contar con su propio gobierno local y el territorio restante es el de la capital nacional, el cual es Nueva Delhi. 

#### Queremos saber cuantos presos quedaron registrados en el año 2001 en el Estado de Assam 

In [12]:
df[(df["year"] == 2005) & (df["state_name"] == "Assam")].shape [0]

10

Se puede decir que el 2005 fue un año donde el Estado de Assam no se presentaron muchos arrestos y podríamos decir que la población tiene buen comportamiento, o que es un Estado pequeño y por eso no hay tantos habitantes.

#### Queremos saber el número máximo registrado durante esos 13 años de personas convictas en la India

In [13]:
df["convicts"].max()

20036

En la India entre el 2001 y el 2013, el máximo de personas convictas en un mismo año y Estado fue de 20036.

#### Ahora queremos saber si ese máximo fue obtenido por hombres o por mujeres

In [14]:
df[["gender", "convicts"]].groupby(["gender"]).max()

convicts
gender          
Female       748
Male       20036

Confirmamos que fue obtenido por hombres, y que el máximo de mujeres convitas que se registró fue de 748. 

#### Queremos saber la cantidad de detenidos durante los 13 años del estudio

In [15]:
df[["year","detenues"]].groupby(["year"]).sum()

detenues
year          
2001      3510
2002      4832
2003      3875
2004      4491
2005      2542
2006      2275
2007      4687
2008      2978
2009      2232
2010      2325
2011      2450
2012      1922
2013      3113

Notamos que no existe una proporcionalidad, porque a medida que avanzan los años ni aumenta ni disminuye la cantidad de detenidos en India. Esto puede deberse a muchos casos hipoteticos, como que un año se cumplieron las leyes con mayor frecuencia, o que llegó un nuevo mandatario y cambio reglas para los estados o incluso el factor pobreza puede aplicar aqui, podriamos decir que en un año la pobreza fue baja, por ende los actos ilicitos bajaron y consecuentemente el número de detenidos. Porque si hay pobreza podría influir en el comportamiento de los ciudadanos, ya que tienen que buscar alguna forma de sobrevivir. 
2002 fue el año que más detenidos tuvo y 2012 el menor.

#### Queremos observar cuantos hombres y mujeres se encuentran detenidos por otras razones y si estan en estados oficiales o no

In [16]:
df[["is_state","gender", "others"]].groupby(["is_state", "gender"]).sum()

others
is_state gender        
0        Female       0
         Male        28
1        Female    8722
         Male     65629

Observamos que en los estados no oficiales no se encuentran registradas mujeres por x razones, en comparacion de los estados que si son oficiales. Y nuevamente notamos que el número de hombres predomina. Como ya sabemos esto puede darse porque los hombres conforman mayor parte de la población

#### Queremos saber según el género de reclusos, cuantos se encuentran bajo juicio

In [17]:
df[["is_state","gender", "under_trial"]].groupby(["is_state", "gender"]).sum()

under_trial
is_state gender             
0        Female         5594
         Male         129384
1        Female       121957
         Male        2869570

Vemos que en los estados oficiales se encuentra un mayor número de personas bajo juicio, esto se da ya que al ser estados oficiales, pues puede existir mayor población, por ende mayor número de reclusos.

#### Queremos saber si los datos tienen alguna relación u proporción entre sí

In [18]:
df.corr()

is_state      year  convicts  under_trial  detenues    others
is_state     1.000000  0.007272  0.111149     0.100153  0.074412  0.029970
year         0.007272  1.000000  0.035191     0.012228 -0.030960 -0.052961
convicts     0.111149  0.035191  1.000000     0.872075  0.345166  0.126571
under_trial  0.100153  0.012228  0.872075     1.000000  0.358825  0.036462
detenues     0.074412 -0.030960  0.345166     0.358825  1.000000  0.465840
others       0.029970 -0.052961  0.126571     0.036462  0.465840  1.000000

Podemos observar que los datos de la base de datos no tienen correlación o es un indice muy muy bajo, es decir, que no existe una proporción directa entre los datos obtenidos en el censo.

#### Queremos saber cual fue la religión que predominó en los convictos de la India durante los 13 que se realizó el censo

In [19]:
df[["religion", "convicts"]].groupby(["religion"]).sum()

convicts
religion           
Christian     56903
Hindu       1050868
Muslim       257572
Others        22990
Sikh          61359

Podemos ver, que durante el 2001 y el 2013 las carceles de la India tuvieron 1.050.868 de convictos que practican el Hinduismo siendo la religión que más predominó, seguida de Islam.

#### Como comentario queriamos decir que la información de la base de datos con la descripción del contexto no es acorde del todo, la descripción nos muestra "convicts" como la orientación religiosa de los prisioneros, pero esta nos muestra es el número de convictos que tiene el estado. 
#### De igual manera "religion" nos muestra la denominación del la persona que sigue a una religión, quien practica el Hinduismo es Hindú, quien practica el Islam es Musulman, Cristianismo es Cristiano y así con los demás, más no dice la religión como tal. 
#### Esto no afecta directamente en los resultados pero si fue algo que pudimos percibir.